ra2ce/analysis/losses/risk_calculation/risk_calculation_base.py

```diff
@@ -62,9 +82,8 @@ def integrate_df_trapezoidal(self) -> np.array:
 
         """
         # convert return periods to frequencies
-        self._to_integrate.columns = [1 / rp for rp in self._to_integrate.columns]
-        # sort columns by ascending frequency
-        self._to_integrate = self._to_integrate.sort_index(axis="columns")
-        values = self._to_integrate.values
-        frequencies = self._to_integrate.columns
+        _to_integrate = self._to_integrate.copy()
+        _to_integrate.columns = sorted(1 / rp for rp in self._to_integrate.columns)
+        values = _to_integrate.values
+        frequencies = _to_integrate.columns
```

Comment:
why doing this? would it not be more logical to just call the `self._get_to_integrate` here?

I would imagine it would make more sense that `_rework_damage_data_` is reworked so that it returns the geodataframe to use in `_to_integrate` having something like this:

```suggestion

        _to_integrate = self._get_damage_data_network()
        _to_integrate.columns = sorted(1 / rp for rp in self._to_integrate.columns)
        values = _to_integrate.values
        frequencies = _to_integrate.columns
```

Where `_get_damage_data_network` replaces the existing `_rework_damage_data` and in each inherited class should be something as:

In the base class:
```python
def _get_damage_data_network(self) -> gpd.GeoDataframe:
  pass
```
In the concrete classes
```python
def _get_damage_data_network(self) -> gpd.GeoDataFrame:
  _to_integrate = self._get_to_integrate() # This method calls the base class
  # whatever needs to be done in the concrete class.
  return _to_integrate
```